### PR TITLE
self-hosted gpu runners

### PR DESCRIPTION
**Pull request by rhodrin**
Move gpu testing over to self-hosted runners.

~Currently testing. Not yet for merge.~

### CODE REVIEW

File: .github/workflows/pytest-gpu.yml [hunk 1]
```diff
@@ -15,85 +11,31 @@ on:
...
-        -adminPassword ${{ secrets.CI_GPU_VM_ADMIN_PASSWORD }}
-
-    - name: set host
-      run: echo ::set-output name=action_host::$(az vm show -d -g ${{ secrets.CI_GPU_RESOURCE_GROUP }} -n gpu_ci_vm_${{ steps.extract_branch.outputs.branch }} --query publicIps -o tsv)
-      id: host
-
-    - name: execute tests on VM
-      uses: fifsky/ssh-action@master
+        # OpenMP5
+        DEVITO_ARCH=clang DEVITO_PLATFORM=nvidiaX pytest --cov --cov-config=.coveragerc --cov-report=xml tests/test_gpu_openmp.py
```
Comment (mloubout) [inline .github/workflows/pytest-gpu.yml:30]: Now that it's a normal action, can these ENV var be set outside of run? Or as a mtrix or something?

Comment (rhodrin) [reply .github/workflows/pytest-gpu.yml:30]: Yes, should be able to achieve something like that.

File: .github/workflows/pytest-gpu.yml [hunk 2]
```diff
@@ -15,85 +11,56 @@ on:
...
+      # Prevent all build to stop if a single one fails
+      fail-fast: false
 
-  # Deploy VM in Azure
-  deploy-vm:
-    runs-on: ubuntu-latest
+      matrix:
+        name: [
+          pytest-gpu-omp,
+          pytest-gpu-omp
```
Comment (mloubout) [inline .github/workflows/pytest-gpu.yml:30]: pytest-gpu-acc

File: .github/workflows/pytest-gpu.yml [hunk 3]
```diff
@@ -15,85 +11,56 @@ on:
...
-              git rev-parse --verify --quiet ${{ steps.extract_branch.outputs.branch }}
-              if [ $? -eq 0 ]; then git branch -D ${{ steps.extract_branch.outputs.branch }}; fi
-              git fetch
-              git checkout ${{ steps.extract_branch.outputs.branch }}
-          fi
-          # Set error trap to catch any failing test
-          err=0
-          trap 'err=1' ERR
-          # OpenMP offloading
-          pgaccelinfo
```
Comment (georgebisbas) [inline .github/workflows/pytest-gpu.yml:88]: can we keep this?

File: .github/workflows/pytest-gpu.yml (hunk 2 above)
Comment (rhodrin) [reply .github/workflows/pytest-gpu.yml]: Ta. Fixed.

File: .github/workflows/pytest-gpu.yml (hunk 3 above)
Comment (rhodrin) [reply .github/workflows/pytest-gpu.yml:88]: Yeah, probably should. It's added back in now.

**Pull request merged**